src/Distribution/Server/Features/Core.hs

```diff
@@ -728,12 +729,15 @@ coreFeature ServerEnv{serverBlobStore = store} UserFeature{..}
       pkginfo <- packageInPath dpath >>= lookupPackageId
       users   <- queryGetUserDb
       let revisions = pkgMetadataRevisions pkginfo
-          revisionToObj rev (_, (utime, uid)) =
-            let uname = userIdToName users uid in
+          revisionToObj rev (cabalFileText, (utime, uid)) =
+            let uname = userIdToName users uid
+                hash = sha256 (cabalFileByteString cabalFileText)
+            in
             Object $ KeyMap.fromList
               [ (Key.fromString "number", Number (fromIntegral rev))
               , (Key.fromString "user", String (Text.pack (display uname)))
               , (Key.fromString "time", String (Text.pack (formatTime defaultTimeLocale "%Y-%m-%dT%H:%M:%SZ" utime)))
+              , (Key.fromString "sha256", String (Text.pack (show hash)))
```

Comment:
Since this type has [a `ToJSON` instance](https://github.com/haskell/hackage-server/blob/e37b78bf3137d3e45852e87904d9120e9ee31dd2/src/Distribution/Server/Features/Security/SHA256.hs#L70), I think it would be better to use that instead of Show. As the behaviour of Show isn't very precisely defined.